core/src/avm2/namespace.rs

```diff
@@ -61,29 +84,79 @@ impl<'gc> Namespace<'gc> {
             .namespaces
             .get(actual_index)
             .ok_or_else(|| format!("Unknown namespace constant {}", namespace_index.0).into());
+        let abc_namespace = abc_namespace?;
 
-        let ns = match abc_namespace? {
-            AbcNamespace::Namespace(idx) => {
-                NamespaceData::Namespace(translation_unit.pool_string(idx.0, context)?)
-            }
-            AbcNamespace::Package(idx) => {
-                NamespaceData::Namespace(translation_unit.pool_string(idx.0, context)?)
-            }
-            AbcNamespace::PackageInternal(idx) => {
-                NamespaceData::PackageInternal(translation_unit.pool_string(idx.0, context)?)
-            }
-            AbcNamespace::Protected(idx) => {
-                NamespaceData::Protected(translation_unit.pool_string(idx.0, context)?)
-            }
-            AbcNamespace::Explicit(idx) => {
-                NamespaceData::Explicit(translation_unit.pool_string(idx.0, context)?)
-            }
-            AbcNamespace::StaticProtected(idx) => {
-                NamespaceData::StaticProtected(translation_unit.pool_string(idx.0, context)?)
+        let index = match abc_namespace {
+            AbcNamespace::Namespace(idx)
+            | AbcNamespace::Package(idx)
+            | AbcNamespace::PackageInternal(idx)
+            | AbcNamespace::Protected(idx)
+            | AbcNamespace::Explicit(idx)
+            | AbcNamespace::StaticProtected(idx)
+            | AbcNamespace::Private(idx) => idx,
+        };
+
+        let mut namespace_name = translation_unit.pool_string(index.0, &mut context.borrow_gc())?;
+
+        // Private namespaces don't get any of the namespace version checks
+        if let AbcNamespace::Private(_) = abc_namespace {
+            return Ok(Self(Gc::new(
+                context.gc_context,
+                NamespaceData::Private(namespace_name),
+            )));
+        }
+
+        // FIXME - what other versioned urls are there?
+        let is_versioned_url = |url: AvmAtom<'gc>| url.as_wstr().is_empty();
+
+        let is_public = matches!(
+            abc_namespace,
+            AbcNamespace::Namespace(_) | AbcNamespace::Package(_)
+        );
+
+        let api_version = if index.0 != 0 {
+            let is_playerglobals = translation_unit
+                .domain()
+                .is_playerglobals_domain(context.avm2);
+
+            let mut api_version = ApiVersion::AllVersions;
+            let stripped = strip_version_mark(namespace_name.as_wstr());
+            let has_version_mark = stripped.is_some();
+            if let Some((stripped, version)) = stripped {
+                assert!(
+                    is_playerglobals,
+                    "Found versioned url {namespace_name:?} in non-playerglobals domain"
+                );
```

Comment:
Can't normal SWFs add a namespace with a version marker? If so, we should warn here, and ignore it (unless FP has a similar `assert!`, or FP actually obeys version markers in non-playerglobals SWFs)

Reply:
That should never happen, so I'd like to get an issue report if any real swf tries to include a version marker